connector/subgraph/src/helpers/agreement.ts

```diff
@@ -41,3 +42,30 @@ export function populateCollateralData(
     collateralRequirement.save()
   }
 }
+
+export function populateVoteCollateralData(
+  proposal: ProposalEntity | null,
+  event: StartVoteEvent
+): void {
+  const convictionVotingApp = ConvictionVotingContract.bind(event.address)
+  const agreementAppAddress = convictionVotingApp.getAgreement()
+  const agreementApp = AgreementContract.bind(agreementAppAddress)
+  const actionData = agreementApp.getAction(proposal.actionId)
+  const collateralRequirementData = agreementApp.getCollateralRequirement(
+    event.address,
+    actionData.value2
+  )
+  const collateralRequirement = new CollateralRequirementEntity(proposal.id)
+
+  collateralRequirement.proposal = proposal.id
+  const collateralTokenId = loadTokenData(collateralRequirementData.value0)
+
+  if (collateralTokenId) {
+    collateralRequirement.token = collateralRequirementData.value0.toHexString()
```

Comment:
Might make more sense

```suggestion
    collateralRequirement.token = collateralTokenId
```